core/src/SYCL/Kokkos_SYCL_Parallel_Reduce.hpp

```diff
@@ -650,11 +625,32 @@ class ParallelReduce<CombinedFunctorReducerType,
     // separately, write the workgroup results back to global memory and recurse
     // until only one workgroup does the reduction and thus gets the final
     // value.
-    if (size > 1) {
-      auto n_wgroups             = (size + wgroup_size - 1) / wgroup_size;
+    if (n_tiles > 0) {
```

Comment:
Is that the else-clause from L596?

Reply:
Yes, it's easier to see when ignoring whitespace changes.

Reply:
Why not using the if-else idiom then?
At the very least comment that this is the else-branch